micropipenv.py

```diff
@@ -640,11 +640,27 @@ def _translate_poetry_dependency(info):  # type: (str) -> str
 
 
 def _poetry2pipfile_lock(
-    only_direct=False, no_default=False, no_dev=False
-):  # type: (bool, bool, bool) -> Dict[str, Any]
+    only_direct=False,
+    no_default=False,
+    no_dev=False,
+    deploy=False,
+):  # type: (bool, bool, bool, bool) -> Dict[str, Any]
     """Convert Poetry files to Pipfile.lock as Pipenv would produce."""
     poetry_lock, pyproject_toml = _read_poetry()
 
+    # If deploy is specified, show a message about the lack of Python versions
+    # parsing capabilities in Micropipenv.
+    # TODO: Implement or use external parser for Python versions in Poetry specification.
+    # See for details: https://github.com/thoth-station/micropipenv/issues/187
+    current_python_version = "{}.{}".format(sys.version_info.major, sys.version_info.minor)
+    wanted_python_version = poetry_lock["metadata"]["python-versions"]
+    message = (
+        "Warning: Currently, Micropipenv is not able to parse complex Python version specifications used by Poetry. "
+        f"Desired version: {wanted_python_version}, current version: {current_python_version}."
+    )
+    level = "warn" if deploy else "debug"
```

Comment:
`warn` [is deprecated](https://docs.python.org/3/library/logging.html#logging.Logger.warning), please use `warning` instead.

```suggestion
    level = "warning" if deploy else "debug"
```